opentelemetry/CHANGELOG.md

```diff
@@ -6,6 +6,7 @@
 - **Breaking** Removed the following public fields and methods from the `SpanBuilder` [#3227][3227]:
   - `trace_id`, `span_id`, `end_time`, `status`, `sampling_result`
   - `with_trace_id`, `with_span_id`, `with_end_time`, `with_status`, `with_sampling_result`
+- **Breaking** Added `#[must_use]` to `opentelemetry::metrics::AsyncInstrumentBuilder`, since unbuilt async instruments will not have their callbacks called. This may cause compile errors for buggy or suspicious code.
```

Comment:
https://doc.rust-lang.org/cargo/reference/semver.html#new-lints

We cannot make breaking changes, but this should be acceptable based on rust guidelines. I suggest to avoid "breaking" work in changelog to avoid scaring users.

My suggestion:

- **Added** `#[must_use]` attribute to `opentelemetry::metrics::AsyncInstrumentBuilder` to add compile time warning when `.build()` is not called on observable instrument builders, preventing silent failures where callbacks are never registered and metrics are never reported.